op-challenger/game/fault/trace/alphabet/provider.go

```diff
@@ -50,7 +53,12 @@ func (ap *AlphabetTraceProvider) GetStepData(ctx context.Context, i types.Positi
 	if traceIndex.Cmp(big.NewInt(int64(len(ap.state)))) >= 0 {
 		return ap.GetStepData(ctx, types.NewPosition(int(ap.depth), big.NewInt(int64(len(ap.state)))))
 	}
-	return BuildAlphabetPreimage(traceIndex, ap.state[traceIndex.Uint64()]), []byte{}, nil, nil
+	key := preimage.LocalIndexKey(4).PreimageKey()
+	// For alphabet output bisection, the state is the local context - that is, the
+	// pre-state l2 block number. So we can just use [ap.state] as the localContext.
+	localContext := common.HexToHash(strings.Join(ap.state, ""))
+	localContextData := types.NewPreimageOracleData(localContext, key[:], nil, 0)
```

Comment:
In https://github.com/ethereum-optimism/optimism/blob/65b163c135e64ecfb8326bea9873c9c47717e233/packages/contracts-bedrock/test/mocks/AlphabetVM2.sol#L20-L34 the `AlphabetVM2` is using the localContext passed to it from the fault dispute game.  In which case I'd have thought it would follow the same algorithm as for output_cannon games and not just be the state.

It turns out this works because `localContext` isn't used for local data with the new contracts - only the old ones.  So I'd just remove this and leave localContext unset. Or maybe even go a step further and remove `LocalContext` from `PreimageOracleData` and hard code the old contract to `NoLocalContext`.

Reply:
Pushed up the change to remove `LocalContext` entirely so we can get this merged.